psimol/main.py

```diff
@@ -491,7 +648,41 @@ def to_mol(self) -> str:
         Returns:
             str: String representing molecule in .mol format
         """
-        pass #TODO
+        
+        # header
+        out = self.name + "\n\n\n"
+
+        unique_bonds = set()
+        for bonds in self.bonds.values():
+            for bond in bonds:
+                if bond not in unique_bonds:
+                    unique_bonds.add(bond)
+
+        # construct atom-to-index dictionary, needed for generating the bond block
+        atom_indices: dict[Atom:int] = { self._atoms[i]:i+1 for i in range(len(self._atoms)) }
+
+        # counts line
+        out += f'{len(self._atoms):3}{len(unique_bonds):3}  0  0  0  0  0  0  0  0999 V2000\n'
+
+        # used to convert between actual atomic charges to .mol file symbolic representations
+        charge_indices = { 0: '  0', 1: '  3', 2: '  2', 3: '  1', -1: '  5', -2: '  6', -3: '  7'  }
+        # atoms
+        for atom in self._atoms:
+            charge = atom.charge if -3 <= atom.charge <= 3 else 0
+            charge = charge_indices[charge]
+            out += f'{atom.x:10.4f}{atom.y:10.4f}{atom.z:10.4f} {atom.symbol:3} 0{charge}  0  0  0  0  0  0  0  0  0\n'
```

Comment:
```suggestion
            out += f'{atom.x:10.4f}{atom.y:10.4f}{atom.z:10.4f} {atom.symbol:3} 0 {charge}  0  0  0  0  0  0  0  0  0\n'
```
We could remove leading spaces from charge_indices values for better readability